### ISSUE TITLE
[JavaScript] Implement new Concept Exercise: array-loops

### ISSUE THREAD

**Issue by SleeplessByte** [exercism/v3#1895]
This issue describes how to implement the `array-loops` concept exercise for the `javascript` track.

## Getting started

**Please please please read the docs before starting.** Posting PRs without reading these docs will be a lot more frustrating for you during the review cycle, and exhaust Exercism's maintainers' time. So, before diving into the implementation, please read up on the following documents:

- [The features of v3](https://github.com/exercism/v3/blob/master/docs/concept-exercises.md).
- [Rationale for v3](https://github.com/exercism/v3/blob/master/docs/rationale-for-v3.md).
- [What are concept exercise and how they are structured?](https://github.com/exercism/v3/blob/master/docs/features-of-v3.md)

Please also watch the following video:

- [The Anatomy of a Concept Exercise](https://www.youtube.com/watch?v=gkbBqd7hPrA).

See the documentation _above_ (general documentation), as well as [How to implement a Concept Exercise in JavaScript](https://github.com/exercism/v3/blob/master/languages/javascript/reference/implementing-a-concept-exercise.md).

## Goal

The goal of this exercise is to teach the student how they can loop / iterate over array items. 

## Learning objectives

- `for`
- `.forEach`
- `while` (maybe?)
- `for of`

## Out of scope

- anything that returns a _new array_ or _mutates_ (transformation)
  - `reduce`
  - `map`
  - `slice`
  - `splice`
- anything that returns a boolean (analysis)
- anything that implicitly loops (arrays)
  - `indexOf`
  - `find`
  - `findIndex`
  - `includes`
  - etc.

## Concepts

- `array-loops`

## Prerequisites

- `arrays`

## Implementing

Use the story from `arrays` / `array-analysis`. 

## Help

You can choose to do this solo-style, or collaborate with multiple people on this. My suggestion would be is to 

1. **first accept this issue by saying "I'd like to work on this"** (you will not get a response from me. Just go with it) and optionally request that someone works with you (and wait for a second person to accept your request).
2. Use this issue to write down what you need (collect reference material) and discuss, if necessary, what to write and what not to write.
3. Draft a PR and request the feedback from at least one other JavaScript member (I recommend just pinging the channel on slack).

**Comment by rishiosaur** [comment]
I'd like to work on this

**Comment by rishiosaur** [comment]
I don't think `while` should be put in this exercise—it doesn't seem right for this kind of thing.

**Comment by SleeplessByte** [comment]
Yeah that's fine. While loops can be a loop exercise.

**Comment by SleeplessByte** [comment]
@rishiosaur How's it going with this? Need help?
